lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_redis_SUITE.erl

```diff
@@ -0,0 +1,412 @@
+}).
+
+-define(COMMON_REDIS_OPTS, #{
+    <<"password">> => <<"public">>,
+    <<"command_template">> => [<<"RPUSH">>, <<"MSGS">>, <<"${payload}">>],
+    <<"local_topic">> => <<"local_topic/#">>
+}).
+
+-define(BATCH_SIZE, 5).
+
+-define(PROXY_HOST, "toxiproxy").
+-define(PROXY_PORT, "8474").
+
+all() -> emqx_common_test_helpers:all(?MODULE).
+
+init_per_suite(Config) ->
+    TestHosts = all_test_hosts(),
+    case emqx_common_test_helpers:is_all_tcp_servers_available(TestHosts) of
+        true ->
+            ProxyHost = os:getenv("PROXY_HOST", ?PROXY_HOST),
+            ProxyPort = list_to_integer(os:getenv("PROXY_PORT", ?PROXY_PORT)),
+            emqx_common_test_helpers:reset_proxy(ProxyHost, ProxyPort),
+            ok = emqx_common_test_helpers:start_apps([emqx_conf]),
+            ok = emqx_connector_test_helpers:start_apps([
+                emqx_resource, emqx_bridge, emqx_rule_engine
+            ]),
+            {ok, _} = application:ensure_all_started(emqx_connector),
+            [
+                {proxy_host, ProxyHost},
+                {proxy_port, ProxyPort}
+                | Config
+            ];
+        false ->
+            {skip, no_redis}
+    end.
+
+end_per_suite(_Config) ->
+    ok = delete_all_bridges(),
+    ok = emqx_common_test_helpers:stop_apps([emqx_conf]),
+    ok = emqx_connector_test_helpers:stop_apps([emqx_rule_engine, emqx_bridge, emqx_resource]),
+    _ = application:stop(emqx_connector),
+    ok.
+
+init_per_testcase(_Testcase, Config) ->
+    ok = delete_all_bridges(),
+    Config.
+
+end_per_testcase(_Testcase, Config) ->
+    ProxyHost = ?config(proxy_host, Config),
+    ProxyPort = ?config(proxy_port, Config),
+    ok = snabbkaffe:stop(),
+    emqx_common_test_helpers:reset_proxy(ProxyHost, ProxyPort),
+    ok = delete_all_bridges().
+
+t_create_delete_bridge(_Config) ->
+    Name = <<"mybridge">>,
+    for_each_conf(fun(Type, _Transport, IsBatch, Conf) ->
+        ?assertMatch(
+            {ok, _},
+            emqx_bridge:create(Type, Name, Conf)
+        ),
+
+        ResourceId = emqx_bridge_resource:resource_id(Type, Name),
+        Health = emqx_resource:health_check(ResourceId),
+
+        ?assertEqual(
+            {ok, connected},
+            Health
+        ),
+
+        RedisType = atom_to_binary(Type),
+        Action = <<RedisType/binary, ":", Name/binary>>,
+
+        RuleId = <<"my_rule_id">>,
+        RuleConf = #{
+            actions => [Action],
+            description => <<>>,
+            enable => true,
+            id => RuleId,
+            name => <<>>,
+            sql => <<"SELECT * FROM \"t/#\"">>
+        },
+
+        %% check export by rule
+        {ok, _} = emqx_rule_engine:create_rule(RuleConf),
+        _ = check_resource_queries(<<"t/test">>, IsBatch),
+        ok = emqx_rule_engine:delete_rule(RuleId),
+
+        %% check export through local topic
+        _ = check_resource_queries(<<"local_topic/test">>, IsBatch),
+
```

Comment:
Suggestion: maybe it's good to also query Redis directly to check that the data is persisted there in the expected format?

Reply:
✅